src/client.js

```diff
@@ -286,34 +285,40 @@ Client.prototype.connect = function(args) {
 	client.save();
 };
 
-Client.prototype.updateToken = function(callback) {
-	var client = this;
-
-	crypto.randomBytes(48, function(err, buf) {
+Client.prototype.generateToken = function(callback) {
+	crypto.randomBytes(48, (err, buf) => {
 		if (err) {
 			throw err;
 		}
 
-		callback(client.config.token = buf.toString("hex"));
+		callback(buf.toString("hex"));
 	});
 };
 
+Client.prototype.updateSession = function(token, ip, request) {
+	const client = this;
+	const agent = UAParser(request.headers["user-agent"] || "");
+
+	client.config.sessions[token] = {
+		lastUse: Date.now(),
+		ip: ip,
+		agent: `${agent.browser.name} ${agent.browser.major} on ${agent.os.name} ${agent.os.version}`,
```

Comment:
Let's not add it in here then, and definitely not as a pre-formatted string.
I did [something here](https://github.com/thelounge/lounge/pull/986/commits/2d4a495721315932518d77fe1ddba49edcd57340) that I think looks really nice in the UI, when both this PR and #986 are ready.